homeassistant/components/garmin_connect/__init__.py

```diff
@@ -86,23 +86,66 @@ def __init__(self, hass, client):
         """Initialize."""
         self.client = client
         self.data = None
+        self._activity_data = None
+        self._body_composition_data = None
 
     @Throttle(MIN_SCAN_INTERVAL)
     async def async_update(self):
         """Update data via library."""
         today = date.today()
 
         try:
-            self.data = self.client.get_stats(today.isoformat())
+            self._activity_data = self.client.get_stats(today.isoformat())
         except (
             GarminConnectAuthenticationError,
             GarminConnectTooManyRequestsError,
         ) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get activity data: %s", err
+            )
             return
         except (GarminConnectConnectionError) as err:
-            _LOGGER.error("Error occurred during Garmin Connect get stats: %s", err)
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get activity data: %s", err
+            )
             return
         except Exception:  # pylint: disable=broad-except
-            _LOGGER.error("Unknown error occurred during Garmin Connect get stats")
+            _LOGGER.error(
+                "Unknown error occurred during Garmin Connect get activity data"
+            )
             return
+
+        try:
+            self._body_composition_data = self.client.get_body_composition(
+                today.isoformat()
+            )["totalAverage"]
+        except (
+            GarminConnectAuthenticationError,
+            GarminConnectTooManyRequestsError,
+        ) as err:
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get body composition data: %s",
+                err,
+            )
+            return
+        except (GarminConnectConnectionError) as err:
+            _LOGGER.error(
+                "Error occurred during Garmin Connect get body composition data: %s",
+                err,
+            )
+            return
+        except KeyError:
```

Comment:
Why is there a `KeyError`. Can `totalAverage` be missing in the result?

Reply:
Normally not, the Ke is included in the response even when user has no body composition data in their profile, I removed the exception, there is always an unknown catch.